lib/http3/server.c

```diff
@@ -1352,8 +1352,22 @@ static int handle_input_expect_headers(struct st_h2o_http3_server_stream_t *stre
 
     h2o_probe_log_request(&stream->req, stream->quic->stream_id);
 
+    int is_connect_udp = 0;
     int is_connect = h2o_memis(stream->req.input.method.base, stream->req.input.method.len, H2O_STRLIT("CONNECT"));
-    int is_connect_udp = h2o_memis(stream->req.input.method.base, stream->req.input.method.len, H2O_STRLIT("CONNECT-UDP"));
+    if (is_connect) {
+        if (stream->req.input.protocol.len != 0) {
+            if ((conn->h3.peer_settings.h3_datagram_rfc) &&
```

Comment:
> if the client is making a draft formatted request, you would expect it to have indicated it is ready to receive draft datagrams and the same for the RFC.

No, a client that does not support datagrams will not send those settings, yet it can still either be making a draft-style request or an rfc-style request.  Such a client will then use the corresponding stream encoding for the datagrams that it sends in the request data stream, and it will parse the response data stream assuming that the server is using the corresponding stream encoding.

Reply:
Ah yes, sending/receiving datagrams over streams does not require this setting. I'll remove this check (you've also corrected this in your suggested mods mentioned elsewhere in this review). Thanks!